jolie/src/jolie/runtime/typing/TypePair.java

```diff
@@ -0,0 +1,84 @@
+package jolie.runtime.typing;
+
+import jolie.lang.NativeType;
+import jolie.runtime.Value;
+import jolie.util.Range;
+
+import java.util.Map;
+
+/**
+ * Created by User on 01.08.2015.
+ */
+public class TypePair extends Type {
+    public Type getT1() {
+        return t1;
+    }
+    public Type getT2() {
+        return t2;
+    }
+
+    private final Type t1;
+    private final Type t2;
+
+    public TypePair(Type t1, Type t2) {
+        this.t2 = t2;
+        this.t1 = t1;
+    }
+
+    @Override
+    public void cutChildrenFromValue(Value value) {
+
+    }
+
+    @Override
+    public NativeType nativeType() {
```

Comment:
Ah, this explains why you have all those "instanceof" checks in the previous code.

I think this shows that we should not have nativeType(), cardinality(), and subTypes() as public methods in Type, since they do not make sense in TypePair. Otherwise we get a leaky abstraction.